state/protocol/badger/mutator.go

```diff
@@ -682,15 +692,30 @@ func (m *FollowerState) epochFallbackTriggeredByFinalizedBlock(block *flow.Heade
 	return blockTriggersEpochFallback, nil
 }
 
+// isFirstBlockOfEpoch returns true if the given block is the first block of a new epoch.
+// Approach: We retrieve the parent block's epoch information. When this block's
+// view exceeds the parent epoch's final view, this block represents the first
+// block of the next epoch.
```

Comment:
I think it is valuable to clarify
```suggestion
// block of the next epoch. There can be multiple (unfinalized) blocks that qualify as 
// the first block of epoch N.  
```

Reply:
Added in https://github.com/onflow/flow-go/pull/3903/commits/d26b84120af20917409a127e0873ca43107b32e3